app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -230,10 +246,29 @@ object BitcoindRpcBackendUtil extends Logging {
           .flatMap { wallet =>
             val runStream: Future[Done] = Source(blockHashes)
               .mapAsync(parallelism = numParallelism) { hash =>
-                bitcoindRpcClient.getBlockRaw(hash)
+                val blockF = bitcoindRpcClient.getBlockRaw(hash)
+                val blockHeaderResultF = bitcoindRpcClient.getBlockHeader(hash)
+                for {
+                  block <- blockF
+                  blockHeaderResult <- blockHeaderResultF
+                } yield (block, blockHeaderResult)
               }
-              .foldAsync(wallet) { case (wallet, block) =>
-                wallet.processBlock(block)
+              .foldAsync(wallet) { case (wallet, (block, blockHeaderResult)) =>
+                val blockProcessedF = wallet.processBlock(block)
+                val executeCallbackF: Future[Wallet] = blockProcessedF.flatMap {
+                  wallet =>
+                    chainCallbacksOpt match {
+                      case None => Future.successful(wallet)
+                      case Some(callback) =>
+                        val f = callback
```

Comment:
where we actually call the `onBlockHeaderConnectedCallback()` for the bitcoind backend. It needs to be done here since we don't have a `chain` module when bitcoind backend is used.